test/e2e/k8s_util.go

```diff
@@ -178,11 +179,77 @@ func DecideProbeResult(stderr string, probeNum int) PodConnectivityMark {
 	return Error
 }
 
+func (k *KubernetesUtils) pingProbe(
+	pod *v1.Pod,
+	podName string,
+	containerName string,
+	dstAddr string,
+	dstName string,
+) PodConnectivityMark {
+	pingCmd := fmt.Sprintf("ping -4 -c 3 -W 1 %s", dstAddr)
+	if strings.Contains(dstAddr, ":") {
+		pingCmd = fmt.Sprintf("ping -6 -c 3 -W 1 %s", dstAddr)
+	}
+	cmd := []string{
+		"/bin/sh",
+		"-c",
+		pingCmd,
+	}
+	log.Tracef("Running: kubectl exec %s -c %s -n %s -- %s", pod.Name, containerName, pod.Namespace, strings.Join(cmd, " "))
+	stdout, stderr, err := k.RunCommandFromPod(pod.Namespace, pod.Name, containerName, cmd)
+	log.Tracef("%s -> %s: error when running command: err - %v /// stdout - %s /// stderr - %s", podName, dstName, err, stdout, stderr)
+	return DecidePingProbeResult(stdout, 3)
+}
+
+// DecidePingProbeResult uses the pingProbe stdout to decide the connectivity.
+func DecidePingProbeResult(stdout string, probeNum int) PodConnectivityMark {
+	// Provide stdout example for different connectivity:
+	// ================== Connected stdout ==================
+	// PING 10.10.1.2 (10.10.1.2) 56(84) bytes of data.
+	// 64 bytes from 10.10.1.2: icmp_seq=1 ttl=64 time=0.695 ms
+	// 64 bytes from 10.10.1.2: icmp_seq=2 ttl=64 time=0.250 ms
+	// 64 bytes from 10.10.1.2: icmp_seq=3 ttl=64 time=0.058 ms
+	//
+	// --- 10.10.1.2 ping statistics ---
+	// 3 packets transmitted, 3 received, 0% packet loss, time 2043ms
+	// rtt min/avg/max/mdev = 0.058/0.334/0.695/0.266 ms
+	// ======================================================
+	// =================== Dropped stdout ===================
+	// PING 10.10.1.2 (10.10.1.2) 56(84) bytes of data.
+	//
+	// --- 10.10.1.2 ping statistics ---
+	// 3 packets transmitted, 0 received, 100% packet loss, time 2037ms
+	// =======================================================
+	// =================== Rejected stdout ===================
+	// PING 10.10.1.2 (10.10.1.2) 56(84) bytes of data.
+	// From 10.10.1.2 icmp_seq=1 Destination Host Prohibited
+	// From 10.10.1.2 icmp_seq=2 Destination Host Prohibited
+	// From 10.10.1.2 icmp_seq=3 Destination Host Prohibited
+	//
+	// --- 10.10.1.2 ping statistics ---
+	// 3 packets transmitted, 0 received, +3 errors, 100% packet loss, time 2042ms
+	// =======================================================
+	countConnected := strings.Count(stdout, "bytes from")
+	countRejected := strings.Count(stdout, "Prohibited")
+	countDropped := probeNum - strings.Count(stdout, "icmp_seq")
```

Comment:
Because I want a special case (one connected one dropped one rejected) still works, although it's uncommon. It could look like this:
```
PING 10.10.1.2 (10.10.1.2) 56(84) bytes of data.
64 bytes from 10.10.1.2: icmp_seq=1 ttl=64 time=0.695 ms
From 10.10.1.2 icmp_seq=2 Destination Host Prohibited

--- 10.10.1.2 ping statistics ---
3 packets transmitted, 1 received, +1 errors, 67% packet loss, time 2042ms
```
`countConnected` should find a keyword that is only in the `connected` reply.
`countDropped` should find a keyword that is in both the `connected` and `rejected` replies.